x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/planner/QueryTranslator.java

```diff
@@ -68,9 +71,26 @@ public static Query toQuery(Expression e, TranslatorHandler handler) {
         for (ExpressionTranslator<?> translator : QUERY_TRANSLATORS) {
             translation = translator.translate(e, handler);
             if (translation != null) {
-                return translation;
+                break;
             }
         }
+        if (translation != null) {
```

Comment:
Wondering if extracting this in a function wouldn't make it more legible -- I guess the only reason to pull it out of the loop is to prevent extra indentation / big loop body -- to have the loop still return in case of translation.